src/khepri_machine.erl

```diff
@@ -1766,24 +1773,23 @@ create_tree_change_side_effects(
                     (_, #{} = _NodeProps)                 -> update;
                     (_, delete)                           -> delete
                 end, Changes0),
-    ProjectionEffects = create_projection_side_effects(
-                          InitialState, NewState, Changes),
-    {NewState1, TriggerEffects} = create_trigger_side_effects(
-                                    InitialState, NewState, Changes),
-    {NewState1, ProjectionEffects ++ TriggerEffects}.
+    SideEffects1 = add_projection_side_effects(
+                     InitialState, NewState, Changes, SideEffects),
+    add_trigger_side_effects(
+      InitialState, NewState, Changes, SideEffects1).
```

Comment:
The trigger-related side effects will appear before the projection-related ones now, won’t they?

Perpaps we should reverse the lines so that trigger-related side effects continue to appear after, in case a trigger function does a read from a projection. What do you think?

Reply:
Hmm yeah, let's try to avoid changing the order effects are processed with this branch. I will also add some `lists:reverse/1` calls in `khepri_machine:apply/3` where these effects are handled so that the order is consistent between transactions and regular commands